contracts/extensions/MultisigPermissions.sol

```diff
@@ -0,0 +1,552 @@
+
+  function getMotion(uint256 motionId) public view returns (Motion memory){
+    return motions[motionId];
+  }
+  function getMotionRoleApprovalCount(uint256 _motionId, ColonyRole _role) public view returns (uint256){
+    return motionRoleApprovalCount[_motionId][_role];
+  }
+
+  function getUserApproval(uint256 _motionId, address _user, ColonyRole _permission) public view returns (bool) {
+    bytes32 approvals = motionApprovals[_motionId][_user];
+    return (approvals >> uint8(_permission)) & bytes32(uint256(1)) == bytes32(uint256(1));
+  }
+
+  function getDomainSkillRoleCounts(uint256 _domainSkillId, uint8 role) public view returns (uint256) {
+    return domainSkillRoleCounts[_domainSkillId][role];
+  }
+
+  function getDomainSkillThreshold(uint256 _domainSkillId) public view returns (uint256) {
+    return domainSkillThreshold[_domainSkillId];
+  }
+
+  function setDomainSkillThreshold(uint256 _domainSkillId, uint256 _threshold) public onlyCoreRoot returns (uint256) {
+    domainSkillThreshold[_domainSkillId] = _threshold;
+  }
+
+  function getDomainSkillRoleThreshold(uint256 _domainSkillId, ColonyRole _role) public view returns (uint256) {
+    // If domain skill threshold explicitly set, use that value
+    if (domainSkillThreshold[_domainSkillId] > 0){
+      return domainSkillThreshold[_domainSkillId];
+    }
+
+    // Otherwise, if global threshold set, return that
+    if (globalThreshold > 0){
+      return globalThreshold;
+    }
+    return domainSkillRoleCounts[_domainSkillId][uint8(_role)]/2 + 1;
+  }
+
+  // Copied role functions
+  function getUserRoles(address who, uint256 where) public view returns (bytes32) {
+    return userDomainRoles[who][where];
+  }
+
+  // Evaluates a "domain proof" which checks that childDomainId is part of the subtree starting at permissionDomainId
+  function validateDomainInheritance(uint256 permissionDomainId, uint256 childSkillIndex, uint256 childDomainId) internal view returns (bool) {
+    if (permissionDomainId == childDomainId) {
+      return childSkillIndex == UINT256_MAX;
+    } else {
+      Domain memory domain = colony.getDomain(permissionDomainId);
+      uint256 childSkillId = colonyNetwork.getChildSkillId(domain.skillId, childSkillIndex);
+      Domain memory childDomain = colony.getDomain(childDomainId);
+      return childSkillId == childDomain.skillId;
+    }
+  }
+
+
+  function setUserRoles(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    address _user,
+    uint256 _domainId,
+    bytes32 _roles
+  ) public
+  {
+    // Validate the permissions proof
+    uint256 colonyDomainCount = colony.getDomainCount();
+    require(_permissionDomainId > 0 && _permissionDomainId <= colonyDomainCount, "multisig-domain-does-not-exist");
+    require(_domainId > 0 && _domainId <= colonyDomainCount, "multisig-domain-does-not-exist");
+    require(validateDomainInheritance(_permissionDomainId, _childSkillIndex, _domainId), "multisig-invalid-domain-inheritance");
```

Comment:
I feel like we could put the first two requires inside of `validateDomainInheritence`